changelogs/fragments/junos_ospf_router_if_fix.yaml

```diff
@@ -0,0 +1,3 @@
+---
+deprecated_features:
+  - Deprecated router_id from ospf-interfaces resource module.
```

Comment:
We are deprecating it right now, but it does not have an alternative (i.e., junos_routing_options). I think we should deprecate it only when a correct option is available. To get past the reported traceback, maybe we just keep the if-condition.